naming/src/main/java/org/jboss/as/naming/ServiceBasedNamingStore.java

```diff
@@ -255,6 +260,16 @@ public Object run() {
         return children;
     }
 
+    @Override
+    public Object lookupLink(Name name) throws NamingException {
+        Object object = lookup(name, false);
+        if (object instanceof LookupBindingManagedReferenceFactory) {
+            return LookupBindingManagedReferenceFactory.class.cast(object).getLink();
```

Comment:
I don't like this at all. This should just work with LinkRef, if should not need a special ManagedReferenceFactory